packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -550,6 +550,9 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     if (searchText.isEmpty) {
       return null;
     }
+    if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase() == searchText) {
+      return currentHighlight;
+    }
     final int index = entries.indexWhere((DropdownMenuEntry<T> entry) => entry.label.toLowerCase().contains(searchText));
```

Comment:
Great job here—I'm glad we aren't searching through all the `entries` to find `currentHighlight` when it was already set in the keyboard event callback.